examples/notebooks/Accessing C Struct Data.ipynb

```diff
@@ -0,0 +1,327 @@
+    "print(\"address of data:\", hex(addr))\n",
+    "result = foo.ctypes(addr, 3)\n",
+    "result"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Manually creating a Numba `Record` type"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Sometimes it is useful to create a `numba.types.Record` type directly.  The easiest way is to use the `Record.make_c_struct()` method. Using this method, the field offsets are calculated from the natural size and alignment of prior fields.\n",
+    "\n",
+    "In the example below, we will manually create the *my_struct* structure from above."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 7,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Record(i1[type=int32;offset=0],f2[type=float32;offset=4],d3[type=float64;offset=8];16;True)"
+      ]
+     },
+     "execution_count": 7,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "from numba import types\n",
+    "\n",
+    "\n",
+    "my_struct = types.Record.make_c_struct([\n",
+    "    # Provides a sequence of 2-tuples i.e. (name:str, type:Type)\n",
+    "    ('i1', types.int32),\n",
+    "    ('f2', types.float32),\n",
+    "    ('d3', types.float64),\n",
+    "])\n",
+    "\n",
+    "my_struct"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Here's another example to demonstrate the offset calculation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "Record(i1[type=int32;offset=0],pad0[type=int8;offset=4],f2[type=float32;offset=8],pad1[type=int8;offset=12],d3[type=float64;offset=16];24;True)"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "padded = types.Record.make_c_struct([\n",
+    "    ('i1', types.int32),\n",
+    "    ('pad0', types.int8),    # padding bytes to move the offsets\n",
+    "    ('f2', types.float32),\n",
+    "    ('pad1', types.int8),    # padding bytes to move the offsets\n",
+    "    ('d3', types.float64),\n",
+    "])\n",
+    "\n",
+    "padded"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Notice how the byte at `pad0` and `pad1` moves the offset of `f2` and `d3`. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A function signature can also be created manually:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "signature: (Record(i1[type=int32;offset=0],f2[type=float32;offset=4],d3[type=float64;offset=8];16;True)*, int64) -> float64\n",
+      "signature matches: False\n"
```

Comment:
Is this right? `False` is recorded on `stdout` WRT this line https://github.com/numba/numba/pull/3661/files#diff-10e250d59b2417305202210f151b5893R302

Reply:
Good catch.  Should be `uintp` not `intp`.

Reply:
fixed.